app/Http/Controllers/RoomController.php

```diff
@@ -14,8 +15,23 @@ public function __construct(JsonResponseService $responseService)
         $this->responseService = $responseService;
     }
 
-    public function all()
+    public function all(Request $request)
     {
+
+        $hidden = ['description'];
+        $search = $request->input('type');
+        $data = Room::with('type:id,name')->whereRelation('type', 'type_id', '=', "$search")->get()->makeHidden($hidden);
+
+        if ($search) {
+
+            if ($data->isEmpty()) {
+                return response()->json($this->responseService->getFormat(
+                    'The selected type is invalid.'));
+            }
```

Comment:
You should use $request->validate() to handle this rather than checking it manually